core/include/optee_msg.h

```diff
@@ -201,6 +201,15 @@ struct optee_msg_arg {
 #define OPTEE_MSG_GET_ARG_SIZE(num_params) \
 	(sizeof(struct optee_msg_arg) + \
 	 sizeof(struct optee_msg_param) * (num_params))
+
+/*
+ * Defines the maximum value of @num_params that can be passed to
+ * OPTEE_MSG_GET_ARG_SIZE without wrapping the result.
+ */
+#define OPTEE_MSG_MAX_NUM_PARAMS	\
+	((SIZE_MAX - sizeof(struct optee_msg_arg)) / \
```

Comment:
Maybe it is worth to use some more meaningful value? Like, `8` for example.

Problem is that `SIZE_MAX` is platform-dependent. So it can cause interesting problems in aarch32/aarch64 interop. Also, right now size of command buffer is partially capped by `SMALL_PAGE_SIZE`, because of check in  ` map_cmd_buffer()`.

Reply:
This is exactly what I had in mind earlier today when asking Jens about it. As you can see on the "fixup", I did change it to use `SMALL_PAGE_SIZE`. This is still quite big, I think I counted that `OPTEE_MSG_MAX_NUM_PARAMS` will be > `190`. I hardly every think you will have anything really bigger than what you said (8), so right now it's more than enough.